File: BE/rempms-communication-service/src/main/java/com/pdev/rempms/communicationservice/model/CommunicationInformation.java
package com.pdev.rempms.communicationservice.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

@Entity
@Data
@Table(name = "communication_information")
public class CommunicationInformation {
    @Id
    @Column(name = "id_communication_information")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "phone_no")
    private String phoneNo;

    @Column(name = "mobile_no")
    private String mobileNo;

    @Column(name = "email")
    private String email;

    @Column(name = "common_status")
    @Length(max = 45)
    private String commonStatus;

    @Embedded
    @AttributeOverrides(value = {
            @AttributeOverride(name = "createdBy", column = @Column(name = "created_by")),
            @AttributeOverride(name = "createdOn", column = @Column(name = "created_on")),
            @AttributeOverride(name = "updatedBy", column = @Column(name = "updated_by")),
            @AttributeOverride(name = "updatedOn", column = @Column(name = "updated_on"))
    })
    private AuditData auditData;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "language_id_language")
    private Language language;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "preferred_communication_id_preferred_communication")
    private PreferredCommunication preferredCommunication;
}

File: FE/rempms-admin-frontend/src/app/model/communication/communicationInformation/CommunicationInformation.ts
import {Language} from "../language/Language";
import {PreferredCommunication} from "../preferredCommunication/PreferredCommunication";

export class CommunicationInformation {
  idCommunicationInformation: string | any
  phoneNo: string | any
  mobileNo: string | any
  email: string | any
  customerName: string | any
  language: Language | any
  preferredCommunication: PreferredCommunication | any
  commInfoIdLanguage: string | any
  commInfoIdPreferredCommunication: string | any
  commInfoCommonStatus: string | any
}
